src/activate/registerCommands.ts

```diff
@@ -117,6 +118,36 @@ const getCommandsMap = ({ context, outputChannel, provider }: RegisterCommandOpt
 		"roo-cline.focusInput": () => {
 			provider.postMessageToWebview({ type: "action", action: "focusInput" })
 		},
+		"roo-cline.commitSuggestion": (() => {
+			let isExecuting = false // Flag to prevent multiple executions
+
+			return async () => {
+				if (isExecuting) {
+					return // Prevent multiple executions
+				}
+
+				try {
+					isExecuting = true
+					await vscode.window.withProgress(
+						{
+							location: vscode.ProgressLocation.Notification,
+							title: "Generating commit suggestion...",
+							cancellable: false,
+						},
+						async () => {
+							// Get the current workspace folder path
+							const workspaceFolders = vscode.workspace.workspaceFolders
+							const cwd = workspaceFolders?.[0]?.uri.fsPath || process.cwd()
```

Comment:
does this pick up the right repo?

Reply:
i think we can make follow up pr for this